lib/generic_esp_32/generic_esp_32.h

```diff
@@ -99,54 +99,288 @@ void sntp_sync_time(struct timeval *tv);
 #ifndef CONFIG_TWOMES_CUSTOM_GPIO
 #define OUTPUT_BITMASK ((1ULL<<RED_LED_ERROR))
 #define INPUT_BITMASK ((1ULL << BOOT))
-
+/**
+ * Initialize the buttons and LEDs on the device, with interrupts on the buttons.
+ */
 void initGPIO();
+
+/**
+ * Check for input of buttons and the duration.
+ * 
+ * If the press duration was more than 10 seconds, erase the flash memory to restart provisioning
+ * otherwise, blink the status LED (and possibly run another task (sensor provisioning?))
+*/
 void buttonPressHandlerGeneric(void *args);
-#endif
+#endif // CONFIG_TWOMES_CUSTOM_GPIO
+
+/**
+ * Blink LEDs to test GPIO.
+ * 
+ * @param args Pass two arguments in uint8_t array:
+ * args[0] = amount of blinks.
+ * args[1] = pin to blink on (LED_STATUS or RED_LED_ERROR).
+ */
 void blink(void *args);
+
+/**
+ * TODO: Not yet able to reverse engineer.
```

Comment:
```suggestion
 * Get types of variadic function arguments.
```

Reply:
Pretty sure this is correct, but it would be great if someone could confirm this.